bootcfg/tlsutil/info.go

```diff
@@ -0,0 +1,41 @@
+package tlsutil
+
+import (
+	"crypto/tls"
+)
+
+// TLSInfo prepares tls.Config's from TLS file inputs.
+type TLSInfo struct {
+	CAFile   string
+	CertFile string
+	KeyFile  string
+}
+
+// ClientConfig returns a tls.Config for client use.
+func (info *TLSInfo) ClientConfig() (*tls.Config, error) {
+	pool, err := NewCertPool([]string{info.CAFile})
+	if err != nil {
+		return nil, err
+	}
+
+	return &tls.Config{
+		MinVersion:         tls.VersionTLS12,
+		InsecureSkipVerify: false,
+		// CA bundle the client should trust when verifying a server
+		RootCAs: pool,
+	}, nil
+}
+
+// ServerConfig returns a tls.Config for server use.
+func (info *TLSInfo) ServerConfig() (*tls.Config, error) {
```

Comment:
If you can get away with extremely limited ciphersuites (no RC4, no CBC, only ECDHE) in these it would be awesome.

```
    CipherSuites: []uint16{
        tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
        tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
    },
```